tests/unit/cli/cli-test.js

```diff
@@ -133,6 +145,29 @@ describe('Unit: CLI', function() {
     td.verify(init(), {ignoreExtraArgs: true, times: 0});
   });
 
+  describe('process signal listeners', function() {
+    var onCommandInterrupt;
+    beforeEach(function() {
+      onCommandInterrupt = td.matchers.isA(Function);
+    });
+
+    it('sets up signal listeners before command started', function() {
+      stubValidateAndRun('serve', function() {
```

Comment:
I think it could be easy to achieve if the `cli` raised some kind of events like "command:beforeRun".
Seems like [`instrumentation`](https://github.com/ember-cli/ember-cli/blob/master/lib/cli/cli.js#L154-L156) does something similar but as I understand it's not supposed to be a notifier.. 

Is it ok(from the aesthetic standpoint) to stub `instrumentation.start('command');` in order to know the moment when we can verify a handler?